narwhal/test-utils/src/lib.rs

```diff
@@ -612,7 +612,8 @@ pub fn mock_certificate_with_epoch(
     epoch: Epoch,
     parents: BTreeSet<CertificateDigest>,
 ) -> (CertificateDigest, Certificate) {
-    let header_builder = HeaderBuilder::default();
+    // TODO(arun): Is there a way to do this without referencing HeaderV1Builder/Header::V1() below?
```

Comment:
@mystenmark ^^

Reply:
why can't we modify HeaderBuilder API? Eventually, you will need access to the protocol config here. If you give the protocol config to HeaderBuilder, it can build the right message type for the current protocol version.

Reply:
I mean HeaderBuilder is generated with #[derive(Builder)], part of derive_builder_macro crate. Maybe we can add a trait to it. But modifying existing signatures seems hard. Or maybe we can hand roll HeaderBuilder instead.

Reply:
Can't derive `Builder` macro on an enum. Would you recommend I create a struct/impl for `HeaderBuilder` and then reference `HeaderV1Builder` in the impl?

Reply:
I think we can do that.

Reply:
Ended up not doing it because piping the protocol config in would require more work. Will merge this and start versioning a few more types and we can come back to this and other improvements with the protocol version number.